File: HackerRank/Week-1/Lonely Integer.py
def lonely_integer(a):
    result = 0
    for num in a:
        result ^= num
    return result

# Reading input
n = int(input())                   
arr = list(map(int, input().split())) 


if len(arr) != n:
    print("No of integers does not match the specified length.")
else:
    print(lonely_integer(arr))

File: HackerRank/Week-1/Lonely_integer.c
#include <stdio.h>

int lonely_integer(int a[], int n) {
    int result = 0;
    for (int i = 0; i < n; i++) {
        result ^= a[i];
    }
    return result;
}

int main() {
    int n;
    scanf("%d", &n);
    int arr[n];

    for (int i = 0; i < n; i++) {
        scanf("%d", &arr[i]);
    }

    if (sizeof(arr)/sizeof(arr[0]) != n) {
        printf("No of integers does not match the specified length.\n");
    } else {
        printf("%d\n", lonely_integer(arr, n));
    }

    return 0;
}
